openlibrary/plugins/upstream/account.py

```diff
@@ -336,9 +336,10 @@ def POST(self):
                 raise olib.code.BadRequest(error)
             expires = 3600 * 24 * 365 if remember.lower() == 'true' else ""
             web.setcookie(config.login_cookie_name, web.ctx.conn.get_auth_token())
-            ol_account = OpenLibraryAccount.get(email=email)
-            if ol_account.get_user().get_safe_mode() == 'yes':
-                web.setcookie('sfw', 'yes', expires=expires)
+            if audit.get('ia_email'):
+                ol_account = OpenLibraryAccount.get(email=audit['ia_email'])
+                if ol_account and ol_account.get_user().get_safe_mode() == 'yes':
+                    web.setcookie('sfw', 'yes', expires=expires)
```

Comment:
Get `ia_email` once, not twice...
```suggestion
            if (
                ia_email := audit.get('ia_email')
                and ol_account := OpenLibraryAccount.get(email=ia_email)
                and ol_account.get_user().get_safe_mode() == 'yes'
            ):
                web.setcookie('sfw', 'yes', expires=expires)
```

Reply:
Today I Learned: You need an extra sent of parens around the `ia_email := audit.get('ia_email')` bit just to make it work properly!  Thanks @mekarpeles